src/libraries/System.Data.Common/tests/System/Data/SqlTypes/SqlXmlTest.cs

```diff
@@ -119,6 +96,82 @@ public void SqlXml_fromXmlReader_CreateReaderTest()
             Assert.Equal(xmlStr, xrdr.ReadOuterXml());
         }
 
+        [Theory]
+        [InlineData("element_whitespace-text.xml")]
+        [InlineData("root_qname.xml")]
+        [InlineData("sample_ecommerce.xml")]
+        [InlineData("sql_batch_request.xml")]
+        [InlineData("sql_batch_response.xml")]
+        [InlineData("sql_datatypes-1.xml")]
+        [InlineData("sql_datatypes-2.xml")]
+        [InlineData("sql_datatypes-3.xml")]
+        [InlineData("xmlns-1.xml")]
+        [InlineData("xmlns-2.xml")]
+        [InlineData("xmlns-3.xml")]
+        [InlineData("xmlns-4.xml")]
+        [InlineData("comments_pis.xml")]
+        [InlineData("element_content_growth.xml")]
+        [InlineData("element_nested-1.xml")]
+        [InlineData("element_nested-2.xml")]
+        [InlineData("element_nested-3.xml")]
+        [InlineData("element_single.xml")]
+        [InlineData("element_stack_growth.xml")]
+        [InlineData("element_tagname_growth.xml")]
+        [InlineData("element_types.xml")]
+        [InlineData("element_whitespace-modes.xml")]
+        public void SqlXml_fromXmlReader_TextXml(string filename)
+        {
+            string filepath = Path.Combine("TestFiles/SqlXml/TextXml", filename);
+
+            using FileStream xmlStream = new FileStream(filepath, FileMode.Open);
+            SqlXml sqlXml = new SqlXml(xmlStream);
+
+            // Reading XML stored as SQL Binary XML will result in using
+            // the XmlTextReader implementation
+            using XmlReader sqlXmlReader = sqlXml.CreateReader();
+
+            // Read to the end to verify no exceptions are thrown
+            while(sqlXmlReader.Read());
```

Comment:
Is there anything more we can validate besides just no exceptions being thrown? e.g. should all of these result in Read returning true at least a certain number of times?  Anything about the state of sqlXml or sqlXmlReader along the way or at the end? etc.  If not, fine, I just want to make sure we're not lulling ourselves into a false sense of security.

Reply:
I didn't want to go down the rabbit hole of writing tests to validate the parsed content of the files, and there weren't any assertions that made sense for _all_ of these test files.

But, I do agree that it's a false sense of security and I was anxious about it to. So here's what I'll do:

- For the SqlBinaryXml files, I'll run them through the reader and pump the content into an XmlTextWriter
- I'll read in the corresponding Text version of the file from disk as well, and compare each result against the baseline

I had kept the two tests separate so that it would be a _little_ more obvious that they are exercising different readers despite the call sites looking the same. But with the above approach, the tests will more obviously need to be different.